src/base/path.cpp

```diff
@@ -52,8 +52,35 @@ const int PATHLIST_TYPEID = qRegisterMetaType<PathList>("PathList");
 
 namespace
 {
+#if defined(Q_OS_WIN)
+    bool isQStrUNCPath(const QString &path)
+    {
+        // strict format of \\host\drive\folder with no forbidden char to avoid conflict with normal path
+        if (path.left(2) != uR"(\\)"_s)
+            return false;
+        const auto lastSlash = path.lastIndexOf(u'\\');
+        if ((lastSlash - 1) <= 2)
+            return false;
+        const QRegularExpression regex {u"[\\0-\\37:?\"*<>|:/]"_s}; // no drive letter allowed C:/
+        return !regex.match(path.data()).hasMatch();
+    }
+#endif
+
     QString cleanPath(const QString &path)
     {
+#if defined(Q_OS_WIN)
+        // QT's cleanPath does not support UNC path
+        if (isQStrUNCPath(path))
```

Comment:
IMO this is full validation which is too expensive here. You should use fast/lightweight checks instead. This function doesn't need validation for constructing paths.